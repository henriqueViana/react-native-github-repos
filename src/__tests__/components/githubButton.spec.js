import React from 'react'
import { shallow, mount } from 'enzyme'
import { TouchableOpacity } from 'react-native'
import { AntDesign } from '@expo/vector-icons'

import GithubButton from '../../components/githubButton'

describe('-----GithubButton-----', () => {
  it('should renders correcty SearchRepository', () => {
    const wrapper = shallow(<GithubButton />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a TouchableOpacity child component', () => {
    const wrapper = shallow(<GithubButton />)

    const childComponent = wrapper.find(TouchableOpacity)
    expect(childComponent).toHaveLength(1)
  })

  it('should render a AntDesign icon child component', () => {
    const wrapper = shallow(<GithubButton />)

    const childComponent = wrapper.find(AntDesign)
    expect(childComponent).toHaveLength(1)
  })

  it('should render a url property', () => {
    const url = 'https://github.com'
    const wrapper = mount(<GithubButton url={url} />)

    expect(wrapper.props().url).toEqual(url)
  })
})