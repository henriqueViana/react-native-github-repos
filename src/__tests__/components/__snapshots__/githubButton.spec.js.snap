// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-----GithubButton----- should renders correcty SearchRepository 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#000",
        "borderWidth": 1,
        "padding": 5,
        "width": 120,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#000"
          name="github"
          size={40}
        />
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 13,
              }
            }
          >
            Abrir no
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            GitHub
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
